source/error_report/src/main.rs

```diff
@@ -0,0 +1,219 @@
+    map.insert("versions".into(), Value::Table(versions));
+    map.insert("verus-output".into(), Value::Table(output));
+    Value::Table(map)
+}
+
+/// Uses the user input file to find the .d file, parse the dependencies,
+/// and write each dependency to the zip file.
+///
+/// Parameters: file_path: a String representation of the path to the input file
+///
+/// Returns:    the names of the .d file and zip file for book-keeping purposes
+pub fn zip_setup(file_path: String) -> (String, String) {
+    let file_name_path = Path::new(&file_path);
+    let temp_file_name =
+        &file_name_path.with_extension(".d").file_name().unwrap().to_string_lossy().to_string();
+    let mut d_file_name = String::new();
+    d_file_name.push_str(&temp_file_name.to_string()[..]);
+    d_file_name = d_file_name[..d_file_name.len() - 2].to_string();
+    d_file_name.push('d');
+    let mut deps = d_to_vec(d_file_name.to_string());
+    deps.push("error_report.toml".to_string());
+    let zip_file_name = write_zip_archive(deps);
+    (d_file_name, zip_file_name)
+}
+
+/// Turns the .d file that lists each of the input files' dependencies
+/// and turns them into a vector of Strings for easier data manipulation
+///
+/// Parameters: file_name: The name of the previously generated .d file
+///
+/// Returns:    a vector containing each dependency of the input file as an
+///             individual string
+fn d_to_vec(file_name: String) -> Vec<String> {
+    let file = File::open(file_name).expect("Couldn't open file!");
+    let mut reader = BufReader::new(file);
+    let mut dependencies = String::new();
+    reader.read_line(&mut dependencies).expect("Could not read the first line");
+    let mut deps = Vec::new();
+    let mut first = true;
+    for dep in dependencies.split_whitespace() {
+        if !first {
+            deps.push(dep.to_string());
+        } else {
+            first = false;
+        }
+    }
+    deps
+}
+
+/// Creates a zip file from a given list of files to compress
+///
+/// Parameters: deps: A vector of strings representing files to be compressed
+///                    (in this context, each file is a dependency of the input)
+///
+/// Returns:    The name of the created zip file
+fn write_zip_archive(deps: Vec<String>) -> String {
+    let local: DateTime<Local> = Local::now();
+    let date = local.to_string();
+    let mut zip_file_name = date[0..19].to_string();
+    zip_file_name.push_str(".zip");
+    zip_file_name.replace_range(10..11, "-");
+    zip_file_name.replace_range(13..14, "-");
+    zip_file_name.replace_range(16..17, "-");
+
+    let path = std::path::Path::new(&zip_file_name);
+    let file = std::fs::File::create(path).unwrap();
+    //let deps = vec!["folder/a.rs", "src/b.rs", "src/c.rs", "src/main.rs"];
+    let mut zip = zip::ZipWriter::new(file);
+    let options = FileOptions::default()
+        .compression_method(zip::CompressionMethod::Stored)
```

Comment:
Actually allow the zip file to compress things, rather than just store things
```suggestion
        .compression_method(zip::CompressionMethod::Bzip2)
```